test/src/test/java/org/corfudb/integration/CorfuReplicationLeaderIT.java

```diff
@@ -0,0 +1,871 @@
+        return CompletableFuture.runAsync(() -> {
+            CorfuRuntime.CorfuRuntimeParameters params = CorfuRuntime.CorfuRuntimeParameters
+                    .builder()
+                    .build();
+            CorfuRuntime activeRuntime =
+                    CorfuRuntime.fromParameters(params)
+                            .setTransactionLogging(true)
+                            .parseConfigurationString(PRIMARY_CORFU_ENDPOINT);
+
+            try {
+                activeRuntime = activeRuntime.connect();
+                CorfuTable<String, Integer> replicationTable = activeRuntime.getObjectsView()
+                        .build()
+                        .setStreamName(REPLICATION_TABLE_NAME)
+                        .setTypeToken(new TypeToken<CorfuTable<String, Integer>>() {
+                        })
+                        .open();
+
+                for (int i = 0; i < iterations; i++) {
+                    log.info("Writer: writing iteration: " + i);
+                    activeRuntime.getObjectsView().TXBegin();
+                    replicationTable.put(String.valueOf(i), i);
+                    activeRuntime.getObjectsView().TXEnd();
+                    Sleep.sleepUninterruptibly(sleepDuration);
+                }
+            } finally {
+                activeRuntime.shutdown();
+            }
+
+        });
+    }
+
+    /**
+     * UUID -> Uuid
+     *
+     * @param uuid
+     * @return
+     */
+    private Uuid convertUUID(UUID uuid) {
+        return Uuid.newBuilder()
+                .setMsb(uuid.getMostSignificantBits())
+                .setLsb(uuid.getLeastSignificantBits())
+                .build();
+    }
+
+    private Tuple<UUID, CorfuInterClusterReplicationServer> runReplicationFuture(int port,
+                                                                                 UUID uuid,
+                                                                                 String pluginConfigPath) {
+        final int duration = 3000;
+        String configuredUUID = UuidUtils.asBase64(uuid);
+        Sleep.sleepUninterruptibly(Duration.ofMillis(duration));
+        String[] args = new String[]{"-m", "--address=localhost", String.valueOf(port),
+                "--plugin=" + pluginConfigPath, "--node-id=" + configuredUUID};
+
+        CorfuInterClusterReplicationServer server =
+                new CorfuInterClusterReplicationServer(args);
+
+        CompletableFuture.runAsync(server);
+
+        return new Tuple<>(uuid, server);
+    }
+
+    private Tuple<UUID, CorfuInterClusterReplicationServer> runReplicationFuture(int port,
+                                                                                 String pluginConfigPath) {
+        UUID uuid = UUID.randomUUID();
+        return runReplicationFuture(port, uuid, pluginConfigPath);
+    }
+
+    @Test
+    public void testLockHolderChangesSourceDuringLogEntrySync() throws Exception {
```

Comment:
can we separate the common code in all tests into a separate method so that the tests are smaller and more readable?